packages/transitive-overlay/src/index.tsx

```diff
@@ -88,14 +89,28 @@ const accessLegFilter = [
 type Props = {
   activeLeg?: Leg;
   transitiveData?: TransitiveData;
+  trimetModeProps: TrimetModeProps;
 };
 
 const TransitiveCanvasOverlay = ({
   activeLeg,
-  transitiveData
+  transitiveData,
+  trimetModeProps
 }: Props): JSX.Element => {
+  const { showRouteArrows = null, ignoreRouteColor = null, trimetTeal = null } =
+    trimetModeProps || {};
   const { current: map } = useMap();
 
+  const calculateColor = segmentData => {
+    let color;
+    if (ignoreRouteColor) {
+      color = segmentData?.route_color || trimetTeal;
+    } else {
+      color = modeColorMap[segmentData.type] || "#008";
+    }
```

Comment:
I don't think this quite works the way we expect it to! What do you think about something like this?
```suggestion
      let color = modeColorMap[segmentData.type] || defaultColorOverride || "#008";
      if (!ignoreRouteColor) {
        color = segmentData?.route_color;
    }
```

Reply:
Also I'm a little bit confused about the `route_color` feature here in general... These are street walking legs! Why are we giving them route colors.

Reply:
Also, if we do simplify this function, I would make it inline again